modules/pubsub/src/main/scala/com/snowplowanalytics/snowplow/enrich/pubsub/Main.scala

```diff
@@ -47,14 +52,22 @@ object Main extends IOApp.WithContext {
   }
 
   def run(args: List[String]): IO[ExitCode] =
-    config.CliConfig.command.parse(args) match {
+    CliConfig.command.parse(args) match {
       case Right(cfg) =>
-        for {
-          _ <- logger.info("Initialising resources for Enrich job")
-          environment <- Environment.make[IO](cfg, executionContext).value
-          exit <- environment match {
-                    case Right(e) =>
-                      e.use { env =>
+        ParsedConfigs.parse[IO](cfg).value.flatMap {
+          case Right(parsed) =>
+            for {
+              _ <- logger.info("Initialising resources for Enrich job")
+              file = parsed.configFile
+              environment = for {
+                              blocker <- Blocker[IO]
```

Comment:
It seems like even more code could be extracted into a common module. Basically the only unique parts here are:

```
rawSource: Stream[F, Payload[F, Array[Byte]]]
mkGoodSink: Resource[F, AttributedByteSink[F]]
piiSink: ...
badSink: ...
```

Maybe I'm overly optimistic, but I hope that eventually all assets would differ only in these four arguments.